test/link.test.js

```diff
@@ -1,74 +1,18 @@
-import { h, app } from "hyperapp"
-import { Route, Link, location } from "../src"
+import { Link } from "../src"
 
-const fakeEvent = {
-  button: 0,
-  currentTarget: { origin: window.location.origin },
-  preventDefault: () => {}
-}
-
-test("Link", done => {
-  const state = {
-    location: location.state
-  }
-
-  const actions = {
-    location: location.actions
-  }
-
-  const main = app(
-    state,
-    actions,
-    (state, actions) =>
-      h("div", {}, [
-        h(Route, {
-          path: "/test",
-          render: () => {
-            h(Link, {
-              to: "/done"
-            })(state, actions).attributes.onclick(fakeEvent)
-          }
-        }),
-        h(Route, {
-          path: "/done",
-          render: () =>
-            h(
-              "h1",
-              {
-                oncreate() {
-                  expect(document.body.innerHTML).toBe(
-                    `<div><h1>Hello</h1></div>`
-                  )
-                  unsubscribe()
-                  done()
-                }
-              },
-              "Hello"
-            )
-        })
-      ]),
-    document.body
-  )
-
-  const unsubscribe = location.subscribe(main.location)
-
-  main.location.go("/test")
-})
-
-test("pass through attributes", () => {
-  const vnode = h(Link, { to: "/path", pass: "through", location })({}, {})
+test('Link passes given attributes except "to" and "location" to underlying element', () => {
+  const vnode = Link({ to: "/path", pass: "through", location })({}, {})
   expect(vnode.attributes.to).toBeUndefined()
   expect(vnode.attributes.location).toBeUndefined()
   expect(vnode.attributes.href).toEqual("/path")
   expect(vnode.attributes.pass).toEqual("through")
 })
 
-test("custom onclick handler", () => {
-  const event = { defaultPrevented: false }
-  const onclick = e => {
-    e.defaultPrevented = true
-  }
-  const vnode = h(Link, { to: "/path", onclick })({}, {})
+test("Calling onclick of VNode transparently calls Link's onclick prop", () => {
+  const onclickProp = jest.fn()
+  const vnode = Link({ onclick: onclickProp })({}, {})
```

Comment:
JSX

Reply:
It is intentional design. There are two reasons:

1. Because of lazy component, the mentioned line equals to the following. Don't you feel strange?

    ```js
    const vnode = (<Link onclick={onclickProp} />)({}, {})
    ```

2. test/link.test.js is for unit testing. `Link` could be used as a component, but it is a pure function.
    - On the other hand, test/index.test.js is for integration testing, so I used JSX there.

Reply:
Okay, makes sense. That would be weird indeed.